src/lib.rs

```diff
@@ -1073,31 +1089,63 @@ impl AssociatedType {
     }
 }
 
+// ===== impl Field =====
+
+impl Field {
+    /// Return a field definition with the provided name and type
+    pub fn new<T>(name: &str, ty: T) -> Self
+    where T: Into<Type>,
+    {
+        Field {
+            name: name.into(),
+            ty: ty.into(),
+            documentation: String::new(),
+            annotation: String::new(),
+        }
+    }
+
+    /// Set field's documentation.
+    pub fn with_documentation(&mut self, documentation: &str) -> &mut Self {
```

Comment:
Would you mind naming this `doc` to match the other functions? If you want, we can bikeshed naming patterns in a separate issue.

Reply:
Sure! Done in 2d8d4b0.